src/ORM/ArrayLib.php

```diff
@@ -274,4 +274,21 @@ public static function iterateVolatile(array &$list)
             }
         }
     }
+
+    /**
+     * Similar to shuffle, but retains the existing keys.
```

Comment:
`retains the existing keys.` is a bit confusing as the keys are what's being shuffled, not the values

Reply:
Would you prefer "retains the existing association between the keys and the values"? Or... something else that you suggest?